daemons/start-daemon.sh

```diff
@@ -44,7 +44,14 @@ then
         echo "Applying patch ${patchfile}"
         
         tmp_bin_file="${TMP_PATCH_DIR}/tmp_bin"
-        filterdiff -i '*/bin/*' "${patchfile}" > ${tmp_bin_file}
+        
+        {
+            filterdiff -i '*/bin/*' "${patchfile}" > ${tmp_bin_file}
+        } ||
+        {
+            echo "Error while filtering patch ${patchfile}/bin. Exiting setup."
+            exit 1
+        }
```

Comment:
Very nice, but why express it like this? Isn’t the following shorter, equivalent, and easier to understand?

```shell
if ! filterdiff -i '*/bin/*' "${patchfile}" > ${tmp_bin_file}; then
    echo "Error while filtering patch ${patchfile}/bin. Exiting setup."
    exit 1
fi
```

Reply:
Amended 

Test script:
```
#!/bin/sh

patchfile="test.patch"
TMP_PATCH_DIR="$(mktemp -d)"
trap 'rm -rf -- "$TMP_PATCH_DIR"' EXIT # Deletes temp dir when script exits
tmp_bin_file="${TMP_PATCH_DIR}/tmp_bin"

if ! filterdiff -i '*/bin/*' "${patchfile}" > ${tmp_bin_file}; then
    echo "Error while filtering patch ${patchfile}/bin. Exiting setup."
    exit 1
fi
```

Output:
```
❯ ./test_script.sh                                                                                                                                                               ─╯
./test_script.sh: line 8: filterdiff: command not found
Error while filtering patch test.patch/bin. Exiting setup.
```